File: 0078-subsets/0078-subsets.java
class Solution {
    
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        backtrack(res, new ArrayList<>(), nums, 0);
        return res;
    }
    
    private void backtrack(List<List<Integer>> list, List<Integer> temp_list, int[] nums, int start){
        list.add(new ArrayList<>(temp_list));
        for(int i = start; i < nums.length; i++){
            temp_list.add(nums[i]);
            backtrack(list, temp_list, nums, i + 1);
            temp_list.remove(temp_list.size()-1);
        }
        
    }
}

File: 0078-subsets/0078-subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        IList<IList<int>> resultList = new List<IList<int>>();
        resultList.Add(new List<int>());

        foreach(int num in nums)
        {
            int size = resultList.Count;
            for(int i = 0; i < size; i++)
            {
                List<int> newSubset = new List<int>(resultList[i]);
                newSubset.Add(num);
                resultList.Add(newSubset);
            }
        }
        return resultList;
    }
}